### PR TITLE
Add support for Dependent Queries to useQuery and usePaginatedQuery

### PR DESCRIPTION
**Pull request by MrLeebo**
### Type: ?? 

Closes: #404 

### What are the changes and their implications? :gear:

Support dependent params in useQuery

### Checklist

- [x] Tests added for changes
- [x] Any added terminal logging uses `packages/server/src/log.ts` (n/a)

### CODE REVIEW

File: packages/core/src/use-query.ts [hunk 1]
```diff
@@ -35,10 +35,15 @@ type RestReturnType<T extends QueryFn, O extends Options<T>> = O['paginated'] ex
...
   queryFn: T,
-  params?: InferUnaryParam<T>,
+  params?: InferUnaryParam<T> | (() => InferUnaryParam<T>),
   options?: O,
 ): [PromiseReturnType<T>, RestReturnType<T, O>] {
-  const queryKey: [string, any] = [(queryFn as any).cacheKey, params]
+  const queryKey: () => [string, any] = () => {
+    return [
+      (queryFn as any).cacheKey,
+      (typeof params === 'function' ? (params as () => InferUnaryParam<T>)() : params) as InferUnaryParam<T>,
```
Comment (flybayer) [inline packages/core/src/use-query.ts:44]: This is all you need to make TS happy for this line:

```suggestion
      typeof params === 'function' ? (params as Function)() : params
```

**Comment by flybayer** [comment]
@MrLeebo I just saw you requested a review from me on this, but it's still in draft mode.

Is this finished or still wip?

**Comment by MrLeebo** [comment]
@flybayer I merged latest from master and applied the same changes to usePaginatedQuery also. I noticed that the `initialData` prop also seems to infer its type from the query function (e.g. expects the initial data to be an async function rather than just the function result) but I don't think that was introduced by this change.

**Comment by MrLeebo** [comment]
I think the github 500 errors are preventing the new commit from showing up.

**Comment by flybayer** [comment]
@MrLeebo ok cool. I'm still not seeing your commit. Maybe try pushing again?

**Comment by MrLeebo** [comment]
Looks like they are there now.

**Comment by flybayer** [comment]
Thank you @MrLeebo!!!

**Pull request merged**